### ISSUE TITLE
Jandex + OpenAPI issue prevent Quickstart-mp example start

### ISSUE THREAD

**Issue by dalexandrov** [oracle/helidon#3690]
## Environment Details
* Helidon Version: 2.3.0
* Helidon MP
* JDK version: 17.0.1
* OS: MacOS
* Docker version (if applicable):

----------

When running simple a `quickstart-mp` application the following issue occurs:

Exception in thread "main" java.lang.reflect.GenericSignatureFormatError: Signature Parse error: Expected Field Type Signature
	Remaining input: org.eclipse.microprofile.openapi.annotations.enums.SchemaType
	at java.base/sun.reflect.generics.parser.SignatureParser.error(SignatureParser.java:124)
....
	at io.helidon.microprofile.cdi.BuildTimeInitializer.<clinit>(BuildTimeInitializer.java:31)
	at io.helidon.microprofile.cdi.ContainerInstanceHolder.fromBuildTimeInitializer(ContainerInstanceHolder.java:58)
	at io.helidon.microprofile.cdi.ContainerInstanceHolder.get(ContainerInstanceHolder.java:50)

It can be fixed by removing `jandex` and excluding Helidon Open API dependancy (with loosing the functionality)

**Comment by tomas-langer** [comment]
Tried with Java 17, 2.4.0 and clean quickstart. Built using maven:
```
2021.12.02 18:07:03 INFO io.helidon.common.LogConfig Thread[main,5,main]: Logging at initialization configured using classpath: /logging.properties
2021.12.02 18:07:04 INFO io.helidon.tracing.tracerresolver.TracerResolverBuilder Thread[main,5,main]: TracerResolver not configured, tracing is disabled
2021.12.02 18:07:04 WARNING io.helidon.microprofile.tracing.TracingCdiExtension Thread[main,5,main]: helidon-microprofile-tracing is on the classpath, yet there is no tracer implementation library. Tracing uses a no-op tracer. As a result, no tracing will be configured for WebServer and JAX-RS
2021.12.02 18:07:04 INFO io.helidon.microprofile.security.SecurityCdiExtension Thread[main,5,main]: Authentication provider is missing from security configuration, but security extension for microprofile is enabled (requires providers configuration at key security.providers). Security will not have any valid authentication provider
2021.12.02 18:07:04 INFO io.helidon.microprofile.security.SecurityCdiExtension Thread[main,5,main]: Authorization provider is missing from security configuration, but security extension for microprofile is enabled (requires providers configuration at key security.providers). ABAC provider is configured for authorization.
2021.12.02 18:07:04 INFO io.helidon.microprofile.server.ServerCdiExtension Thread[main,5,main]: Registering JAX-RS Application: HelidonMP
2021.12.02 18:07:04 INFO io.helidon.webserver.NettyWebServer Thread[nioEventLoopGroup-2-1,10,main]: Channel '@default' started: [id: 0xe338945e, L:/[0:0:0:0:0:0:0:0]:8080]
2021.12.02 18:07:04 INFO io.helidon.microprofile.server.ServerCdiExtension Thread[main,5,main]: Server started on http://localhost:8080 (and all other host addresses) in 1308 milliseconds (since JVM startup).
2021.12.02 18:07:04 INFO io.helidon.common.HelidonFeatures Thread[features-thread,5,main]: Helidon MP 2.4.0 features: [CDI, Config, Fault Tolerance, Health, JAX-RS, Metrics, Open API, REST Client, Security, Server, Tracing]
```

Can you please post full output? We have this error on `master` (due to chnages in OpenAPI spec), but it should not be on release.

**Comment by dalexandrov** [comment]
Fixed in 2.4.0
